core/src/main/java/org/apache/accumulo/core/data/Key.java

```diff
@@ -107,6 +107,17 @@ public Key(Text row) {
   }
 
   /**
+   * Creates a key with the specified row, empty column family, empty column qualifier, empty column visibility, timestamp {@link Long#MAX_VALUE}, and delete
+   * marker false.
```

Comment:
I think adding a Javadoc comment to your method additions (all of them, not just this one) that explicitly state that they copy the provided byte arrays is important. This can be a big performance impact -- you could point to the `Key(byte[] row, byte[] cf, byte[] cq, byte[] cv, long ts, boolean deleted, boolean copy)` constructor which does allow users to control whether or not the constructor will do a copy of the provided byte arrays.